src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -314,12 +314,12 @@ internal static object DefaultToType(IConvertible value, Type targetType, IForma
         // Conversions to Boolean
         public static bool ToBoolean([NotNullWhen(true)] object? value)
         {
-            return value == null ? false : ((IConvertible)value).ToBoolean(null);
+            return value != null && ((IConvertible)value).ToBoolean(null);
```

Comment:
These could be `value is IConvertible convertible && convertible.ToBoolean(null)`, yes?

Reply:
This could be a change in behavior?

Reply:
Certainly a change in code generation, arguably for the worse: https://csharp.godbolt.org/z/nrM7df3YP

Reply:
What makes you say it's "worse"?

The code is nearly identical, just without the early exit for `null`.